code/modules/projectiles/mounted.dm

```diff
@@ -0,0 +1,282 @@
+		if(!operator.interactee)
+			stack_trace("/obj/machinery/mounted/interact(mob/user) called by user [human_user] with an operator with a null interactee: [operator].")
+			operator = null //this shouldn't happen, but just in case
+		to_chat(human_user, "<span class='warning'>Someone's already controlling it.</span>")
+		return TRUE
+	if(human_user.interactee) //Make sure we're not manning two guns at once, tentacle arms.
+		to_chat(human_user, "<span class='warning'>You're already busy!</span>")
+		return TRUE
+	if(issynth(human_user) && !CONFIG_GET(flag/allow_synthetic_gun_use))
+		to_chat(human_user, "<span class='warning'>Your programming restricts operating heavy weaponry.</span>")
+		return TRUE
+
+	visible_message("[icon2html(src, viewers(src))] <span class='notice'>[human_user] mans the [src]!</span>",
+		"<span class='notice'>You man the gun!</span>")
+
+	operator = human_user
+
+	return ..()
+
+/obj/machinery/mounted/examine(mob/user) //Let us see how much ammo we got in this thing.
+	. = ..()
+	gun.examine(user)
+
+/obj/machinery/mounted/attackby(obj/item/I, mob/user, params) //This handles reloading the gun, if its in acid cant touch it.
+	. = ..()
+
+	if(!ishuman(user))
+		return
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return
+
+	reload(user, I)
+
+///Repairs gun
+/obj/machinery/mounted/welder_act(mob/living/user, obj/item/I)
+	if(user.do_actions)
+		return FALSE
+
+	var/obj/item/tool/weldingtool/WT = I
+
+	if(!WT.isOn())
+		return FALSE
+
+	for(var/obj/effect/xenomorph/acid/A in loc)
+		if(A.acid_t == src)
+			to_chat(user, "You can't get near that, it's melting!")
+			return TRUE
+
+	if(obj_integrity == max_integrity)
+		to_chat(user, "<span class='warning'>[src] doesn't need repairs.</span>")
+		return TRUE
+
+	if(user.skills.getRating("engineer") < SKILL_ENGINEER_METAL)
+		user.visible_message("<span class='notice'>[user] fumbles around figuring out how to repair [src].</span>",
+		"<span class='notice'>You fumble around figuring out how to repair [src].</span>")
+		var/fumbling_time = 5 SECONDS * ( SKILL_ENGINEER_METAL - user.skills.getRating("engineer") )
+		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_BUILD))
+			return TRUE
+
+	user.visible_message("<span class='notice'>[user] begins repairing damage to [src].</span>",
+	"<span class='notice'>You begin repairing the damage to [src].</span>")
+	playsound(loc, 'sound/items/welder2.ogg', 25, TRUE)
+
+	if(!do_after(user, 5 SECONDS, TRUE, src, BUSY_ICON_FRIENDLY))
+		return TRUE
+
+	if(!WT.remove_fuel(2, user))
+		to_chat(user, "<span class='warning'>Not enough fuel to finish the task.</span>")
+		return TRUE
+
+	user.visible_message("<span class='notice'>[user] repairs some damage on [src].</span>",
+	"<span class='notice'>You repair [src].</span>")
+	playsound(loc, 'sound/items/welder2.ogg', 25, TRUE)
+	repair_damage(120)
+	update_icon_state()
```

Comment:
```suggestion
	update_icon()
```

Reply:
fucking dammit why? i just changed it from that before